doc/arch/embedded-content-in-govspeak.md

```diff
@@ -0,0 +1,99 @@
+## Decision Record: Embedded content in Govspeak
+
+GOV.UK uses its Markdown dialect, govspeak, as the source of content for the
+site. At present the conversion from govspeak to HTML is done in the publishing
+apps.
+
+One of the features of govspeak is that it can embed other items, eg contacts,
+in the markdown, and they are output as rendered HTML. This means that
+presentation information is included in the data sent to the frontend apps,
+rather than being determined by those frontend apps themselves, which makes
+it harder than necessary to iterate on the design.
+
+To improve this, govspeak rendering will be moved to be a responsibility of the
+publishing API, and embedded content will be separated out so that the details
+are sent to frontends rather than the rendered output.
+
+
+## Description
+
+A Whitehall editor edits the body of a document and chooses a contact from a
+javascript widget populated by the linkables endpoint of publishing-api. This
+causes text containing a contact ID to be inserted into the document body:
+
+```
+Some text
+
+[Contact:1234]
+
+More text
+```
+
+The short ID continues to be used here as it is more friendly for editors,
+some of whom know their frequently-used IDs.
+
+When sending to publishing-api, Whitehall converts this to an `embed` element
+with the relevant content_id:
+
+```
+Some text
+
+[embed:2b4d92f3-f8cd-4284-aaaa-25b3a640d26c]
+
+More text
+```
+
+and additionally adds that content_id to the the relevant element of the
```

Comment:
I don't think so. At the very least, this will only affect the draft stack, and only momentarily. Content is not pushed to the live content store until after publish, but put_links happens before that - if for some reason it doesn't, then the HTML will simply remain as a comment.